test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.Transforms;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.ML.Scenarios
+{
+    public partial class ScenariosTests
+    {
+        private class TestData
+        {
+            public float[] a;
+            public float[] b;
+        }
+        [Fact]
+        public void TensorflowTransformMatrixMultiplicationTest()
+        {
+            var model_location = GetDataPath("model_matmul");
+            using (var env = new TlcEnvironment(seed: 1, conc: 1))
+            {
+                // Pipeline
+                var loader = ComponentCreation.CreateDataView(env,
+                    new List<TestData>(new TestData[] { new TestData() { a = new[] { 1.0f, 2.0f,
+                                                                                     3.0f, 4.0f },
+                                                                         b = new[] { 1.0f, 2.0f,
+                                                                                     3.0f, 4.0f } },
+                        new TestData() { a = new[] { 2.0f, 2.0f,
+                                                     2.0f, 2.0f },
+                                         b = new[] { 3.0f, 3.0f,
+                                                     3.0f, 3.0f } } }));
+
+                var trans = new TensorflowTransform(env, loader, model_location, "c", "a", "b");
```

Comment:
I would hope that we don't need to explicitly declare this but can instead read it from model.  We should try to auto-map the IDV column names to TF input names.  Only when we can't map them should we require some user input.

Reply:
But this will completely preclude any compile-time validation.

---
In reply to: [209317833](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209317833) [](ancestors = 209317833)

Reply:
I don't see how listing strings without type information provides compile time validation.  Sure if the pigsty tuples are used then I'd expect those to show up and just flow through the pipeline and work assuming remappings don't need to happen.  I'd only expect a caller to explicitly list column names here if they need to change them from tuples that have already been defined in the data.

Reply:
Sonoma does this FYI, you can look at how they did it.

---
In reply to: [209413718](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209413718) [](ancestors = 209413718,209375203,209374964,209329418,209324258,209317833)